extensions/ql-vscode/src/extension.ts

```diff
@@ -748,9 +748,13 @@ async function activateWithInstalledDistribution(
     );
   ctx.subscriptions.push(qlConfigurationListener);
 
+  void extLogger.log("Initializing CodeQL language server.");
+  const languageClient = createLanguageClient(qlConfigurationListener);
+
```

Comment:
moving this up a couple hundred lines is perhaps the most risky change in this PR — this changes the initialization order of things because we now inject a reference to the language client into thes cli server. It seems to work fine for me locally.